bundle-workflow/python/assemble_workflow/bundle.py

```diff
@@ -38,6 +38,8 @@ def install_plugins(self):
         for plugin in self.plugins:
             print(f'Installing {plugin.name}')
             self.install_plugin(plugin)
+            if plugin.name == 'k-NN':
```

Comment:
I'm confused why this is necessary. Why isn't the installation of the plugin itself doing this copy?

Reply:
There are a couple reasons here that I can see of why this is currently installed separately.

1. This lib is not included as part of the plugin zip distribution.  It is a separate component in addition to the zip. 
2. If it were included in the zip, we need to update the[ install cli ](https://github.com/opensearch-project/OpenSearch/blob/19e54d662488a183c1388e66cfe7463a24c6aa7c/distribution/tools/plugin-cli/src/main/java/org/opensearch/plugins/InstallPluginCommand.java#L856) to support this.

Reply:
There is also an additional step for setting up performance analyzer in our existing script.  This one-off logic should be included during the install step as well.
https://github.com/opensearch-project/opensearch-build/blob/main/release/tar/linux/opensearch-tar-build.sh#L147

Reply:
Let's find a more generic way to execute these? Maybe along with build.sh we can add an install.sh?

Reply:
Not sure if plugin team can handle these using post/pre-installation script. Example: https://github.com/opensearch-project/performance-analyzer/blob/main/packaging/deb/postinst

Reply:
`install.sh` would be a good workaround until the OpenSearch plugin installer provides a way for plugin zips to include and install native libraries.

For more "why is this necessary", take a look at https://github.com/opensearch-project/opensearch-build/blob/main/release/tar/linux/opensearch-tar-install.sh to see how LD_LIBRARY_PATH is set to consume the native library. TL;DR: OpenSearch plugins don't really have a good way to include native libraries.

Reply:
Interesting! I opened https://github.com/opensearch-project/opensearch-plugins/issues/75